app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -111,9 +111,9 @@ def rhsm_index
     #api :POST, "/hypervisors", N_("Update the hypervisors information for environment")
     #desc 'See virt-who tool for more details.'
     def hypervisors_update
-      #TODO
-      cp_response, _ = System.register_hypervisors(@environment, @content_view, params.except(:controller, :action, :format))
-      render :json => cp_response
+      results = sync_task(::Actions::Katello::Host::Hypervisors, @environment, @content_view,
+                          params.except(:controller, :action, :format))
+      render :json => results.output.try(:results)
     end
```

Comment:
a)  the task execution needs to be wrapped in:
```
      User.as_anonymous_admin do
      end
```
block

b) results.output.try(:results) won't work.  You want  results.output[:results].  You might also rename results to 'action' or 'task' to be more clear

I was getting an exception due to both of these things